From: "Borislav Petkov (AMD)" <bp@alien8.de>
Subject: UBSAN: use the kernel panic message markers
Date: Mon, 18 Dec 2023 14:53:39 +0100

Use the same splat markers as panic does for easier matching by external
tools scanning kernel dmesg for splats.

Link: https://lkml.kernel.org/r/20231218135339.23209-1-bp@alien8.de
Signed-off-by: Borislav Petkov (AMD) <bp@alien8.de>
Cc: Andrey Ryabinin <ryabinin.a.a@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 lib/ubsan.c |    7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

--- a/lib/ubsan.c~ubsan-use-the-kernel-panic-message-markers
+++ a/lib/ubsan.c
@@ -204,8 +204,8 @@ static void ubsan_prologue(struct source
 {
 	current->in_ubsan++;
 
-	pr_err("========================================"
-		"========================================\n");
+	pr_warn(CUT_HERE);
+
 	pr_err("UBSAN: %s in %s:%d:%d\n", reason, loc->file_name,
 		loc->line & LINE_MASK, loc->column & COLUMN_MASK);
 
@@ -215,8 +215,7 @@ static void ubsan_prologue(struct source
 static void ubsan_epilogue(void)
 {
 	dump_stack();
-	pr_err("========================================"
-		"========================================\n");
+	pr_warn("---[ end trace ]---\n");
 
 	current->in_ubsan--;
 
_
